targets/wasm_exec.js

```diff
@@ -298,9 +298,22 @@
 
 					// func finalizeRef(v ref)
 					"syscall/js.finalizeRef": (v_ref) => {
-						// Note: TinyGo does not support finalizers so this should never be
-						// called.
-						console.error('syscall/js.finalizeRef not implemented');
+						// Note: TinyGo does not support finalizers so this is only called
+						// for one specific case, by js.go:jsString. and can/might leak memory.
+						const id = mem().getUint32(unboxValue(v_ref), true);
+						// Note that this if is so far seemingly never true. Someone should investigate why.
+						if (this._goRefCounts?.[id] !== undefined) {
+							this._goRefCounts[id]--;
+							if (this._goRefCounts[id] === 0) {
+								const v = this._values[id];
+								this._values[id] = null;
+								this._ids.delete(v);
+								this._idPool.push(id);
+							}
+						} else {
+							// Log as a hint and reminder that something is probably off.
+							console.log("syscall/js.finalizeRef: unknown id", id);
```

Comment:
```suggestion
							console.error("syscall/js.finalizeRef: unknown id", id);
```

Reply:
This is what I meant.

Reply:
That would get back to previous state where every invocation would error, which is what I'm trying to avoid, I think it's fine as low level log "for developers", as a reminder it needs more work,

For users of a wasm module I think they shouldn't worry about it imo
maybe a compromise is a Warning but even that would probably make me want to not produce wasm using tinygo

Reply:
> That would get back to previous state where every invocation would error

I suppose I do not understand the purpose of this PR in that case. I thought you wanted to see the error in logs from some of your previous comments.

Reply:
The purpose is two(*) fold:

- avoid showing errors on otherwise functional or seemingly functional wasm js
- use the code in the 3 year old issue that might apply in some cases
- yet not just assume the id or ref object exist when they don't, allowing further investigation 


*: off by one error

Reply:
in other words see the not quite expected case in the log, but not as error